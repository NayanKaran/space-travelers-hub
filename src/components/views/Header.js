import logo from '../../logo.png';
import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
  <header>
    <img src={logo} className="App-logo" alt="logo" />
    <nav>
      <NavLink className={({ isActive }) => (isActive ? 'active-link' : undefined)} to="/">Rockets</NavLink>
      <NavLink className={({ isActive }) => (isActive ? 'active-link' : undefined)} to="/missions">Missions</NavLink>
      <NavLink className={({ isActive }) => (isActive ? 'active-link' : undefined)} to="/my-profile">My Profile</NavLink>
    </nav>
  </header>
);

export default Header;