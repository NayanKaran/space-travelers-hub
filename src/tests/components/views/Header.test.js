import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';
import Header from '../../../components/views/Header';

describe('Testing Header component', () => {
  it('Testing with snapshot', () => {
    const tree = renderer
      .create(
        <Router>
          <Header />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
