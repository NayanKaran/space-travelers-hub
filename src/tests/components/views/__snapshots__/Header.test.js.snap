// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Header component Testing with snapshot 1`] = `
<header>
  <img
    alt="logo"
    className="App-logo"
    src="logo.png"
  />
  <nav>
    <a
      aria-current="page"
      className="active-link"
      href="/"
      onClick={[Function]}
    >
      Rockets
    </a>
    <a
      href="/missions"
      onClick={[Function]}
    >
      Missions
    </a>
    <a
      href="/my-profile"
      onClick={[Function]}
    >
      My Profile
    </a>
  </nav>
</header>
`;
